collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java

```diff
@@ -315,14 +244,145 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
             }
         });
         mSearchButton = (ImageButton) findViewById(R.id.search_button);
-        mSearchButton.setOnClickListener(new OnClickListener() {
+        mSearchButton.setOnClickListener(this);
 
-            @Override
-            public void onClick(View v) {
-                executeSearch();
+        // Initialize credentials and service object.
+        mCredential = GoogleAccountCredential.usingOAuth2(
+                getApplicationContext(), Collections.singletonList(DriveScopes.DRIVE))
+                .setBackOff(new ExponentialBackOff());
+
+        HttpTransport transport = AndroidHttp.newCompatibleTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+        mDriveService = new com.google.api.services.drive.Drive.Builder(
+                transport, jsonFactory, mCredential)
+                .setApplicationName("ODK-Collect-Drive")
+                .build();
+
+        getResultsFromApi();
+    }
+
+    /*
+     * Attempt to call the API, after verifying that all the preconditions are
+     * satisfied. The preconditions are: Google Play Services installed, an
+     * account was selected and the device currently has online access. If any
+     * of the preconditions are not satisfied, the app will prompt the user as
+     * appropriate.
+     */
```

Comment:
Hmm, I thought it would be helpful with some of the OAuth stuff but I wasn't finding the best reference to it. It was just coming up on a generic "set up an Android project" page. If you have a page with the code samples and some explanation, that would be a good link to have. I think you may already have linked to something like that in the sheets feature.